### PR TITLE
41 update display stops use case to return name of stops

### PR DESCRIPTION
**Pull request by prash-red**
**Summary of Changes:**
- [x] Modify the `display_stops` use-case to include stop names in the response sent to the front-end. The updated response now contains stop names, stop tags, and stop locations.
- [x] Adjust the front-end to display stop names when a user hovers over a stop.
- [x] Implement functionality on the front end to showcase detailed information about stops.
- [x] Introduce new test files for the `entity` folder and the `display_stops` use-case.

**Related Issues:**
- Addresses the issue #41, which emphasizes the need to enhance the `display_stops` use-case to provide stop names.
- Resolves issue #34, which is related to the broader improvements in stop information presentation.

These changes aim to improve the user experience by enriching the stop information displayed on the front-end, making it more informative and user-friendly. The adjustments in the use-case and front-end components are aligned with the project's goals and contribute to a more comprehensive display of stop details.

Please review the changes, and feel free to provide any feedback or suggestions for further enhancements.

WhatsApp Image 2023-11-26 at 12 35 06 AM

### CODE REVIEW

**Review by KuzeyOzturac — APPROVED** [review]
The logic here is clear and well-organized. But I would wait until someone else also approves the changes to merge.

File: backend/src/main/java/com/example/backend/use_case/display_stops/StopObject.java
Comment (KuzeyOzturac) [inline backend/src/main/java/com/example/backend/use_case/display_stops/StopObject.java:1]: Why have you introduced a new Stop object class?

File: backend/src/main/java/com/example/backend/use_case/display_stops/StopsOutputData.java
Comment (prash-red) [reply backend/src/main/java/com/example/backend/use_case/display_stops/StopsOutputData.java:1]: That it is the type of object that the front-end expects

File: backend/src/test/java/com/example/backend/entity/LocationTest.java
Comment (prash-red) [reply backend/src/test/java/com/example/backend/entity/LocationTest.java:1]: I feel like its better to change it, but it does not really matter in this case as we dont have a lot of test cases.

File: backend/src/test/java/com/example/backend/entity/RouteDirectionTest.java
Comment (prash-red) [reply backend/src/test/java/com/example/backend/entity/RouteDirectionTest.java:1]: Same as the previous comment

File: backend/src/test/java/com/example/backend/entity/RouteTest.java
Comment (prash-red) [reply backend/src/test/java/com/example/backend/entity/RouteTest.java:1]: Same as the previous comment

File: backend/src/test/java/com/example/backend/entity/StopTest.java
Comment (prash-red) [reply backend/src/test/java/com/example/backend/entity/StopTest.java:1]: Will change

File: backend/src/test/java/com/example/backend/use_case/display_stops/StopObjectTest.java
Comment (prash-red) [reply backend/src/test/java/com/example/backend/use_case/display_stops/StopObjectTest.java:1]: Will change

File: frontend/pages/HomePage.tsx
Comment (prash-red) [reply frontend/pages/HomePage.tsx:1]: The title page is just for aesthetics, The project doesnt really need a title page.

File: backend/src/test/java/com/example/backend/entity/LocationTest.java
Comment (KuzeyOzturac) [reply backend/src/test/java/com/example/backend/entity/LocationTest.java:1]: It would make a small difference anyways. If you cannot get BeforeAll to work, leave it as BeforeEach.

**Review by nsaroiu — APPROVED** [review]
Ngl the addition of StopObjects is very regarded, but if it truly does make a significant difference in efficiency for the frontend, then I suppose it is not regarded enough to warrant a change.

File: backend/src/main/java/com/example/backend/use_case/display_stops/StopsOutputData.java
Comment (prash-red) [reply backend/src/main/java/com/example/backend/use_case/display_stops/StopsOutputData.java:19]: ⢀⡴⠑⡄⠀⠀⠀⠀⠀⠀⠀⣀⣀⣤⣤⣤⣀⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀ 
⠸⡇⠀⠿⡀⠀⠀⠀⣀⡴⢿⣿⣿⣿⣿⣿⣿⣿⣷⣦⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⠑⢄⣠⠾⠁⣀⣄⡈⠙⣿⣿⣿⣿⣿⣿⣿⣿⣆⠀⠀⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⢀⡀⠁⠀⠀⠈⠙⠛⠂⠈⣿⣿⣿⣿⣿⠿⡿⢿⣆⠀⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⢀⡾⣁⣀⠀⠴⠂⠙⣗⡀⠀⢻⣿⣿⠭⢤⣴⣦⣤⣹⠀⠀⠀⢀⢴⣶⣆ 
⠀⠀⢀⣾⣿⣿⣿⣷⣮⣽⣾⣿⣥⣴⣿⣿⡿⢂⠔⢚⡿⢿⣿⣦⣴⣾⠁⠸⣼⡿ 
⠀⢀⡞⠁⠙⠻⠿⠟⠉⠀⠛⢹⣿⣿⣿⣿⣿⣌⢤⣼⣿⣾⣿⡟⠉⠀⠀⠀⠀⠀ 
⠀⣾⣷⣶⠇⠀⠀⣤⣄⣀⡀⠈⠻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡇⠀⠀⠀⠀⠀⠀ 
⠀⠉⠈⠉⠀⠀⢦⡈⢻⣿⣿⣿⣶⣶⣶⣶⣤⣽⡹⣿⣿⣿⣿⡇⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⠀⠀⠀⠉⠲⣽⡻⢿⣿⣿⣿⣿⣿⣿⣷⣜⣿⣿⣿⡇⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⣿⣷⣶⣮⣭⣽⣿⣿⣿⣿⣿⣿⣿⠀⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⠀⠀⣀⣀⣈⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠇⠀⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⠀⠀⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠃⠀⠀⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⠀⠀⠀⠹⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡿⠟⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀ 
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠉⠛⠻⠿⠿⠿⠿⠛⠉

**Pull request merged**
